### PR TITLE
Check maxTokenizationLineLength in monarchLexer

### LINKED ISSUE #3025: [Bug] Setting `maxTokenizationLineLength` has no effect
### Reproducible in vscode.dev or in VS Code Desktop?

- [X] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [ ] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Monaco Editor Playground Code

```typescript
monaco.editor.create(document.getElementById('container'), {
  value: "[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]",
  language: 'javascript',
  maxTokenizationLineLength: 4,
});
```

### Actual Behavior

The line gets tokenized and highlighted.

**Note:** when hovering over the line it actually says that tokenization was skipped.

### Expected Behavior

Tokenization should be skipped.

### Additional Context

The default `maxTokenizationLineLength` value is `20_000`. When I paste a longer line into https://vscode.dev it isn't highlighted, however it gets highlighted in the Monaco playground.

Since Monaco handles highlighting separately, my guess would be that this option is not taken into account.

The on-hover messages is added [here](https://github.com/microsoft/vscode/blob/7104e8fcf43136054261575c8ec311e0238a8fbb/src/vs/editor/contrib/hover/browser/markdownHoverParticipant.ts#L75-L82) (for both Monaco and VSCode) and the TextMate tokenization is skipped [here](https://github.com/microsoft/vscode/blob/7104e8fcf43136054261575c8ec311e0238a8fbb/src/vs/workbench/services/textMate/browser/abstractTextMateService.ts#L440-L443).

### PR DESCRIPTION
**Pull request by Lazyuki**
This PR fixes https://github.com/microsoft/monaco-editor/issues/3025

The issue is described in the issue linked above, but basically when `monaco-editor` was parsing and tokenizing lines, it was using `monarch` tokenizer instead of `TextMate` tokenizer which was already checking the `maxTokenizationLineLength` option. So I added the `maxTokenizationLineLength` check in `monarch` tokenizer as well.

### CODE REVIEW

**Comment by Lazyuki** [comment]
Sorry for the delay, I've finally been signed off by my employer @alexdima

**Comment by Lazyuki** [comment]
Could I get reviews? This bug is freezing up the UIs and there's no workaround other than fixing it in the source.

**Comment by Lazyuki** [comment]
@alexdima My bad, I messed up the rebase bu it should be fixed now. I ran the tests locally and they all passed.

**Comment by Lazyuki** [comment]
This PR is ready for review @alexdima

**Comment by Lazyuki** [comment]
Hi @hediet you seem to be assigned to the original bug report issue. Could you run the CI? This bug is making the UI unresponsive for 5+ seconds sometimes.

**Comment by Lazyuki** [comment]
@alexdima @hediet could I please get a review?

**Comment by Lazyuki** [comment]
It seems like the `Hygiene and Layering` job is stuck

**Comment by Lazyuki** [comment]
Looks like the recently added `Code OSS` check is failing on all the PRs made in the past 3 days.

**Comment by Lazyuki** [comment]
@alexdima hello could you run the CI?

**Review by alexdima — APPROVED** [review]

File: src/vs/editor/standalone/common/monarch/monarchLexer.ts [hunk 1]
```diff
@@ -424,6 +426,16 @@ export class MonarchTokenizer implements languages.ITokenizationSupport {
 				emitting = false;
 			}
 		});
+		this._maxTokenizationLineLength = this._configurationService.getValue<number>('editor.maxTokenizationLineLength', {
+			overrideIdentifier: this._languageId
+		});
+		this._configurationService.onDidChangeConfiguration(e => {
```
Comment (alexr00) [inline src/vs/editor/standalone/common/monarch/monarchLexer.ts:432]: Should this listener be disposed at some point?

**Review by alexr00 — APPROVED** [review]

File: src/vs/editor/standalone/common/monarch/monarchLexer.ts (hunk 1 above)
Comment (alexdima) [reply src/vs/editor/standalone/common/monarch/monarchLexer.ts:432]: Good point! I'll do a follow-up PR.

**Comment by alexdima** [comment]
Thank you!

**Pull request merged**